### PR TITLE
feat(api): add vercel ai helper to node sdk

### PR DESCRIPTION
**Pull request by hekike**
I'm still debating if this should be in SDK or just an example.
I can see an argument for both. What do you think?

### CODE REVIEW

File: api/client/node/next.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  }: {
+    model: TiktokenModel
+    prompts: string[]
+  },
+  openAIStreamCallbacks: OpenAIStreamCallbacksWithUsage
+) {
+  // Tiktoken is an optional dependency, so we require it conditionally
+  if (!encodingForModel) {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { encodingForModel: encodingForModel_ } = require('js-tiktoken')
```
Comment (tothandras) [inline api/client/node/next.ts:29]: `import` and `require` shouldn't be mixed

File: api/client/node/next.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      for (const content of prompts) {
+        const tokens = enc.encode(content)
+        promptTokens += tokens.length
+      }
+
+      if (typeof openAIStreamCallbacks?.onStart === 'function') {
+        return openAIStreamCallbacks.onStart()
+      }
+    },
+    onToken: (content) => {
```
Comment (tothandras) [inline api/client/node/next.ts:50]: To keep it consistent
```suggestion
    onToken (content) {
```

File: api/client/node/test/next.test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+              done()
+            },
+          }
+        )
+
+        streamCallbacks.onStart && streamCallbacks.onStart()
+        streamCallbacks.onToken && streamCallbacks.onToken('Hello ')
+        streamCallbacks.onToken && streamCallbacks.onToken(' World!')
+        streamCallbacks.onFinal && streamCallbacks.onFinal('Hello World!')
```
Comment (tothandras) [inline api/client/node/test/next.test.ts:33]:
```suggestion
        streamCallbacks.onStart?.()
        streamCallbacks.onToken?.('Hello ')
        streamCallbacks.onToken?.(' World!')
        streamCallbacks.onFinal?.('Hello World!')
```

File: api/client/node/test/next.test.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    it('should return usage', () =>
+      new Promise<void>((done) => {
+        const streamCallbacks = createOpenAIStreamCallback(
+          {
+            model,
+            prompts,
+          },
+          {
+            onUsage: (usage) => {
+              expect(usage).toEqual({
```
Comment (tothandras) [inline api/client/node/test/next.test.ts:19]: You might need to wrap this into a try/catch and reject the promise?

File: api/client/node/test/next.test.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            },
+          }
+        )
+
+        streamCallbacks.onStart && streamCallbacks.onStart()
+        streamCallbacks.onToken && streamCallbacks.onToken('Hello ')
+        streamCallbacks.onToken && streamCallbacks.onToken(' World!')
+        streamCallbacks.onFinal && streamCallbacks.onFinal('Hello World!')
+
+        expect(true).toBe(true)
```
Comment (tothandras) [inline api/client/node/test/next.test.ts:35]: Why do you need this? :)

File: api/client/node/test/next.test.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+          },
+          callbacks
+        )
+
+        streamCallbacks.onStart && streamCallbacks.onStart()
+        streamCallbacks.onToken && streamCallbacks.onToken('Hello ')
+        streamCallbacks.onToken && streamCallbacks.onToken(' World!')
+        streamCallbacks.onCompletion &&
+          streamCallbacks.onCompletion('Hello World!')
+        streamCallbacks.onFinal && streamCallbacks.onFinal('Hello World!')
```
Comment (tothandras) [inline api/client/node/test/next.test.ts:70]:
```suggestion
        streamCallbacks.onStart?.()
        streamCallbacks.onToken?.('Hello ')
        streamCallbacks.onToken?.(' World!')
        streamCallbacks.onCompletion?.('Hello World!')
        streamCallbacks.onFinal?.('Hello World!')
```

File: api/client/node/test/next.test.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+        expect(true).toBe(true)
```
Comment (tothandras) [inline api/client/node/test/next.test.ts:72]: Same

**Review by tothandras — CHANGES_REQUESTED** [review]

File: api/client/node/test/next.test.ts (hunk 4 above)
Comment (hekike) [reply api/client/node/test/next.test.ts:19]: It exists this way too

File: api/client/node/next.ts (hunk 1 above)
Comment (hekike) [reply api/client/node/next.ts:29]: ES6 imports are async by default. You can't do synchronous `import`, only with `require`. As this factory is synchronous, I don't see a way to use `import`. Btw I don't know any reason why mixing would impact a Node.js application.

**Pull request merged**